### ISSUE TITLE
Prüfwürfe?

### ISSUE THREAD

**Issue by starsailor78** [kreuvf/roll20-character-sheets#40]
Hallo - ich hoffe, ich bin hier richtig... Bei einer Roll20-Sitzung mit DSA-4.1-Template werden seit einiger Zeit Nah- und Fernkampfangriffe mit 2 Würfeln dargestellt... ich tippe darauf, dass es sich bei dem 2. um eine Trefferzonenangabe oder eine Art Prüfwurf handelt, was etwas verwirrend ist.
Kann man das entweder a) abstellen oder wenigstens b) die getroffene Zone ausspucken lassen?
Eigenschafts- und Talentwürfe werden korrekt mit 1 bzw. 3 Würfeln dargestellt.

Vielen Dank für die tolle Arbeit!

**Comment by kreuvf** [comment]
> Hallo - ich hoffe, ich bin hier richtig... 

Ob ihr wirklich richtig stehe, seht ihr, wenn das Licht angeht: :bulb:

> Bei einer Roll20-Sitzung mit DSA-4.1-Template werden seit einiger Zeit Nah- und Fernkampfangriffe mit 2 Würfeln dargestellt... ich tippe darauf, dass es sich bei dem 2. um eine Trefferzonenangabe oder eine Art Prüfwurf handelt, was etwas verwirrend ist.

Dieses "seit einiger Zeit" ist dann so um die drei Jahre? ^^

> Kann man das entweder a) abstellen oder wenigstens b) die getroffene Zone ausspucken lassen?

Nein, das lässt sich nicht abstellen. Das ist auch kein Fehler und fällt nur auf, wenn die 3D-Würfel genutzt werden. Der Hintergrund ist folgender: Bei Nah- und Fernkampfwürfen ist es möglich eine 1 oder 20 zu würfeln. Um eine entsprechende glückliche Attacke/Parade oder einen Patzer zu erhalten, wird aber ein Prüfwurf benötigt. Nun lässt Roll20 es nicht zu, dass ein Charakterbogen von sich aus auf Basis eines Wurfergebnisses erneut würfelt, wenn es nötig wird. Daher würfelt der Bogen grundsätzlich mit zwei Würfeln. Leider ist es nicht möglich diese Würfel irgendwie unterschiedlich einzufärben, um sie auseinanderhalten zu können. Es bleibt daher nur der Blick in den Chat.

**Comment by starsailor78** [comment]
Es ist uns erst gestern aufgefallen, obwohl wir schon ein paar Wochen mit Roll 20 unterwegs sind... kann aber auch sein, dass wir eher den Würfel-Assistenten benutzt haben anstatt direkt die Nahkampf-Attacken zu klicken (haben erst allmählich alle Charakterbögen eingegeben).

Trotzdem danke für die Info und keep up the great work! Oh, und wenn man noch einen Wunsch anführen darf (nur nice to have): Im Reiter "Magie" kann man Rituale zwar eingeben, aber nicht direkt drauf würfeln... wir parken die erst mal im Bereich "Gaben" oder Metatalente, was auch funktioniert.

**Comment by kreuvf** [comment]
> Im Reiter "Magie" kann man Rituale zwar eingeben, aber nicht direkt drauf würfeln... wir parken die erst mal im Bereich "Gaben" oder Metatalente, was auch funktioniert.

Eines der fernen Ziele ist es, auf Dauer jeden Bereich so umfangreich zu überarbeiten wie den Fernkampf mit dem Fernkampfrechner. Magie im Allgemeinen und Rituale im Besonderen sind leider noch eher rudimentär im Bogen abgebildet.

**Comment by kreuvf** [comment]
Ich habe nochmal etwas herumüberlegt und denke, dass man das mit den Würfeln lösen könnte. Den zweiten Wurf müsste man dann zwar manuell ausführen, aber ich vermute, dass das von Benutzern der 3D-Würfel sogar erwünscht ist, oder? ^^

**Comment by starsailor78** [comment]
Das wäre ein kleiner Preis und eine Lösung sehr willkommen :-)

**Comment by Koizumi85** [comment]
Hm. Also du willst quasi eigene Roll-Buttons für die Prüfwürfe hinzufügen? Das wären dann aber ganz schön viele, oder?
Wir bräuchten ja dann quasi für jede Sonderfertigkeit nen eigenen Prüfwurf machen. Weiß nicht ob das nicht ein bisschen viel wäre.
Also Hammerschlag-Prüfwurf, Schildattacke-Prüfwurf, ...
Und dann müssten wir ja eigentlich einen Prüfwurf für Glückliche Attacken und einen für Patzer jeweils dazu packen...

**Comment by kreuvf** [comment]
> Hm. Also du willst quasi eigene Roll-Buttons für die Prüfwürfe hinzufügen? Das wären dann aber ganz schön viele, oder?
> Wir bräuchten ja dann quasi für jede Sonderfertigkeit nen eigenen Prüfwurf machen. Weiß nicht ob das nicht ein bisschen viel wäre.

Nein, abhängig davon, ob die Einstellung "Prüfwürfe manuell würfeln" aktiv ist oder nicht, wird der Wurf hinter dem Würfel geändert und auch ein anderes Roll Template benutzt. Den Wurf kann man in einem Attribut speichern. Habe es noch nicht probiert, aber in meiner Vorstellung funktioniert das xD Sollte das nur über Extrawürfelbuttons gehen, wäre das den Aufwand nicht wert.

**Comment by Koizumi85** [comment]
Ah okay. Falls man das "Prüfwürfe manuell würfeln" aktiviert hat, sollten wir im Prüfwurf eventuell einen Text wie 
"Kritisch, falls Prüfwurf <= 16 gelingt" hinzufügen. Dann müssen die Spieler nicht selbst rumrechnen gegen welchen Wert Sie die Prüfwürfe würfeln müssen.

**Comment by kreuvf** [comment]
> Ah okay. Falls man das "Prüfwürfe manuell würfeln" aktiviert hat, sollten wir im Prüfwurf eventuell einen Text wie
> "Kritisch, falls Prüfwurf <= 16 gelingt" hinzufügen. Dann müssen die Spieler nicht selbst rumrechnen gegen welchen Wert Sie die Prüfwürfe würfeln müssen.

Ich vermute, dass das nicht gehen wird. In der Regel ist der Prüfwurf eine Wiederholung des ersten Wurfes mit allen Erschwernissen, ABER (das typische DSA-Aber) eine 20 im Fernkampf wird ohne Modifikatoren, also nur auf den FK-Wert, geprüft. Diese Information würde ich in das geänderte Roll Template reinschreiben, ansonsten (= Nahkampf) nur "Wurf mit den gleichen Modifikatoren wiederholen!". Natürlich kann das Sheet auch nicht erkennen, wenn ein zweiter Wurf in Folge 1 oder 20 war, sodass die Spieler selbst checken müssen, dass sie kein drittes Mal zu würfeln brauchen.

**Comment by Koizumi85** [comment]
Ah. Ich hab auch Unsinn geschrieben. wir sollten den Text nicht im Prüfwurf sondern bei der normalen Attacke reinschreiben.
Und da wissen wir ja den eigentlichen AT-Wert und wir wissen die Modifikatoren. Also können wir ja anzeigen auf welchen Wert gewürfelt werden muss, oder?

**Comment by kreuvf** [comment]
> Ah. Ich hab auch Unsinn geschrieben. wir sollten den Text nicht im Prüfwurf sondern bei der normalen Attacke reinschreiben.
> Und da wissen wir ja den eigentlichen AT-Wert und wir wissen die Modifikatoren. Also können wir ja anzeigen auf welchen Wert gewürfelt werden muss, oder?

Im Roll Template kann man *nicht* rechnen, nur vorher berechnete Werte anzeigen. Ich sehe aber, dass es `eff_wert` gibt, was an und für sich schon eine Krücke für die Nutzung der Roll Templates ist. Den Wert könnte man dann aber anzeigen, da dieser mit dem Würfelergebnis verglichen wird (siehe die viel zu lang verschachtelten Roll Templates dazu ...). Kommt aber definitiv nicht mehr in den PR für Montag xD

**Comment by Koizumi85** [comment]
Ja. Das kann man ja dann in die nächste Version mit reinnehmen. Ich denke da werden in nächster Zeit noch einige folgen :)

**Comment by kreuvf** [comment]
> Ich denke da werden in nächster Zeit noch einige folgen :)

Das "befürchte" ich auch ;D

**Comment by Koizumi85** [comment]
😂 Ich freu mich drauf

**Comment by Koizumi85** [comment]
Also Ich hab mir das hier gerade mal angeschaut. Was man auf jeden Fall machen kann ist, dass man den Prüfwurf in ein Attribut schreibt. Wenn man dann in das Attribut bspw. stattdessen einfach eine "0" schreibt, dann wird der zweite 3D-Würfel nicht geworfen.

Was etwas komplizierter wird, ist die Geschichte, dass man im Chat noch anzeigt, ab welchem Wert der Prüfwurf gelungen wäre oder nicht. Das liegt aber nicht daran, dass die Berechnung kompliziert wäre, sondern einfach nur daran dass das Template wirklich, wirklich hart verschachtelt ist. Das sollte man auf jeden Fall irgendwie neu aufbauen und vereinfachen. Auch wenn man dann vielleicht nicht so viele tolle Farben hat oder so. Auf diese Weise ist es wirklich schwer nachzuvollziehen...
Ich werd mich mal noch etwas tiefer in dieses Monstrum einarbeiten und versuchen eine gute Lösung dafür zu finden.

**Comment by kreuvf** [comment]
> Also Ich hab mir das hier gerade mal angeschaut. Was man auf jeden Fall machen kann ist, dass man den Prüfwurf in ein Attribut schreibt. Wenn man dann in das Attribut bspw. stattdessen einfach eine "0" schreibt, dann wird der zweite 3D-Würfel nicht geworfen.

S. unten. Ich baue die Sachen lieber getrennt ein als noch mehr Kram in die bestehende Struktur reinzufrickeln, weil das so schon kaum wartbar ist xD

> Was etwas komplizierter wird, ist die Geschichte, dass man im Chat noch anzeigt, ab welchem Wert der Prüfwurf gelungen wäre oder nicht. Das liegt aber nicht daran, dass die Berechnung kompliziert wäre, sondern einfach nur daran dass das Template wirklich, wirklich hart verschachtelt ist. Das sollte man auf jeden Fall irgendwie neu aufbauen und vereinfachen. Auch wenn man dann vielleicht nicht so viele tolle Farben hat oder so. Auf diese Weise ist es wirklich schwer nachzuvollziehen...
> Ich werd mich mal noch etwas tiefer in dieses Monstrum einarbeiten und versuchen eine gute Lösung dafür zu finden.

Die vielen tollen Farben sind nur ein Abfallprodukt davon, dass das korrekt sämtliche Fälle abdeckt. Das hat beim Debuggen/Entwickeln auch geholfen, um schnell zu sehen, ob das Template korrekt arbeitet. Daher bin ich dafür neue Roll Templates zu nutzen, die dann nur zwischen 1, gelungen, misslungen und 20 unterscheiden müssen, ohne dass man da was schachteln muss.

**Comment by Koizumi85** [comment]
Ich hab mir da grad mal paar Gedanken zu gemacht und denke wenn wir die Templates in dieser Form aufbauen, dann sollte es mit relativ wenig Schachtelung etc. möglich sein die Informationen zum Prüfwurf unterzubringen und auch eine farbliche Unterscheidung zwischen den einzelnen Ergebnissen zu erzielen, da die Unterscheidung tatsächlich nur an ganz spezifischen Stellen stattfindet. Was hältst du davon? Dann würde Ich da mal ein paralleles neues Template für anlegen, mit dem man das dann umsetzen und testen kann.

![image](https://user-images.githubusercontent.com/7482685/90340729-98681f00-dffa-11ea-875a-c162f8a9d3eb.png)

**Comment by kreuvf** [comment]
> ![image](https://user-images.githubusercontent.com/7482685/90340729-98681f00-dffa-11ea-875a-c162f8a9d3eb.png)

Ich verstehe das Ziel dahinter nicht so ganz ^.^ Möchtest du grundsätzlich, dass die AT/PA-Würfe anders aussehen, ist das Ziel ein einheitlicheres Aussehen, wenn man die Würfe ohne Prüfwurf dazunimmt oder was ganz anderes? Für die Würfe ohne Prüfwurf, würde ich beim bekannten Schema bleiben: hellblau/grün/gelb/orange für 1/gelungen/misslungen/20. Darunter dann die Erklärung. Ich bin mir sicher, dass es verschmerzbar ist, wenn das (imho sehr schöne) Lila und das Rot nicht mehr in Erscheinung treten. Ich verstehe auch nicht, wieso wir dem Spieler erklären müssen, bei welchem W20-Wurf er das schafft. Der Spieler soll einfach nochmal AT/PA würfeln und damit hat es sich dann erledigt. Bei FK steht im Hinweis, dass es eine unmodifizierte Probe ist, bei AT/PA, dass die Probe nochmal mit allen Modifikatoren gewürfelt werden muss wie vorher - welche das in Summe waren, sieht man aber am Wurf.

**Comment by Koizumi85** [comment]
Also die Farben waren jetzt generell einfach so gepickt. Ich hab es auf Anhieb nicht geschafft ne 1 zu würfeln um zu schauen wie die Farben tatsächlich aussehen und war zu faul mir die Farbcodes aus dem CSS rauszusuchen.
Dass Ich Ihnen sagen möchte wann Sie es schaffen hat einfach nur den Grund, dass viele wahrscheinlich nicht nochmal Attacke würfeln sondern wahrscheinlich eher einen normalen W20 Wurf würfeln wollen. Dafür ist die Angabe dann glaube Ich sinnvoll.
Ansonsten hat das ganze zwei Ziele:
- Die unterschiedlichen Würfelergebnisse sehen derzeit sehr sehr unterschiedlich aus (was zumindest bei mir dazu führt, dass Ich jedes mal wieder die Hälfte davon lesen muss um zu erkennen was es jetzt genau war)
- Das ganze Template so zu vereinfachen, dass man keine Raketenwissenschaft studiert haben muss um zu verstehen was da passiert 😂

**Comment by Koizumi85** [comment]
So... erste Arbeiten abgeschlossen. Sieht jetzt so aus:

![image](https://user-images.githubusercontent.com/7482685/90343654-dffaa500-e012-11ea-8847-a5f32e2b8af3.png)

Stand:
- Prüfwurf kann per Attribut an-/ausgeschaltet werden (Entsprechende Checkbox noch nicht eingebaut)
- Ohne Prüfwurf werden die entsprechenden Abschnitte nicht angezeigt, sondern stattdessen die Aufforderung den Prüfwurf zu würfeln und die Aussage über den nötigen Wurf
- Das Template ist VIEL einfacher. Ich habe es von mehr als 200 Zeilen auf ca. 60 oder 70 Zeilen Umfang reduziert

Noch zu tun:
- Eventuell die Anordnung der Texte/Zahlen nochmal anpassen
- Bei Elementen die einen Tooltip haben die Strichellinie untendrunter zu setzen
- Oben genannte Checkbox hinzufügen um das Attribut zu setzen

**Comment by starsailor78** [comment]
Ich bin begeistert und beeindruckt von der Arbeit, die hier reingesteckt wird - vielen Dank! Könnt ihr absehen, wann die Änderungen in Roll20 implementiert und sichtbar werden? Oder ist das kein Automatismus und ich muss ein neues Sheet laden oder so...

**Comment by kreuvf** [comment]
Ist noch nicht im Roll20, aber @Koizumi85 und ich haben uns schon auf einen Weg verständigt, wie wir das ins Sheet aufnehmen :)

**Comment by starsailor78** [comment]
Frohes Neues Jahr! Eine Frage noch (weiß nicht, ob man dafür einen eigenen Thread öffnen sollte): bei den Waffen können INI-Boni (z. B. Säbel +1) oder -Mali (Orknase -1) vermerkt werden. Ist nicht vorgesehen, dass diese in den INI-Wurf eingerechnet werden? (Habe es mit der als aktiv markierten Waffe aus dem Nahkampf-Reiter getestet).

**Comment by kreuvf** [comment]
@starsailor78 Ich möchte schon mal vorankündigen, dass ein entsprechendes Feature voraussichtlich noch im Juni 2023 als Testversion (Betafeature) allgemein verfügbar wird. :)

**Comment by kreuvf** [comment]
@starsailor78, für den Fall, dass du den DSA4.1-Charakterbogen auf Roll20 noch benutzt: Das ist in einer Betaversion aktuell für einen Teil der Knöpfe verfügbar. :)
